src/mscorlib/src/System/Text/StringBuilder.cs

```diff
@@ -999,6 +999,124 @@ void ISerializable.GetObjectData(SerializationInfo info, StreamingContext contex
             return this;
         }
 
+        public StringBuilder Append(StringBuilder value)
+        {
+            Contract.Ensures(Contract.Result<StringBuilder>() != null);
+
+            if (value != null && value.Length > 0)
+            {
+                AppendCore(value, 0, value.Length);
+            }
+            return this;
+        }
+
+        public StringBuilder Append(StringBuilder value, int startIndex, int count)
+        {
+            if (startIndex < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(startIndex), Environment.GetResourceString("ArgumentOutOfRange_GenericPositive"));
+            }
+            if (count < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(count), Environment.GetResourceString("ArgumentOutOfRange_GenericPositive"));
+            }
+            Contract.Ensures(Contract.Result<StringBuilder>() != null);
+            Contract.EndContractBlock();
+
+            if (value == null)
+            {
+                if (startIndex == 0 && count == 0)
+                {
+                    return this;
+                }
+                throw new ArgumentNullException(nameof(value));
+            }
+            if (count > value.Length - startIndex)
+            {
+                throw new ArgumentOutOfRangeException(nameof(count), Environment.GetResourceString("ArgumentOutOfRange_IndexLength"));
+            }
+
+            if (count > 0)
+            {
+                AppendCore(value, startIndex, count);
+            }
+            return this;
+        }
+
+        private unsafe void AppendCore(StringBuilder value, int startIndex, int count)
+        {
+            Contract.Assert(value != null, "StringBuilder is null");
+            Contract.Assert(count > 0, "count should be more zero");
+            Contract.Assert(startIndex >= 0 && startIndex <= value.Length - count, "startIndex and count out of range");
```

Comment:
Yes it is `private void AppendCore(StringBuilder value, int startIndex, int count)` methods and it does not make double checking after `public Append(StringBuilder value, int startIndex, int count)` method.